src/blue/CallbacksSnippets.sol

```diff
@@ -72,10 +76,11 @@ contract CallbacksSnippets is IMorphoSupplyCollateralCallback, IMorphoRepayCallb
     }
 
     function onMorphoLiquidate(uint256 repaidAssets, bytes calldata data) external onlyMorpho {
-        (uint256 toSwap, MarketParams memory marketParams) = abi.decode(data, (uint256, MarketParams));
-        uint256 returnedAmount = swapper.swapCollatToLoan(toSwap);
-        require(returnedAmount > repaidAssets); // Add logic for gas cost threshold for instance
-        ERC20(marketParams.loanToken).approve(address(swapper), returnedAmount);
+        (address collateralToken) = abi.decode(data, (address));
+
+        ERC20(collateralToken).approve(address(swapper), type(uint256).max);
```

Comment:
I don't think it's a good practice to infinite approve a swapper contract, so i would recommend approving only the amount required
Not a big deal